src/converse-register.js

```diff
@@ -390,6 +415,9 @@
                     if (ev && ev.preventDefault) { ev.preventDefault(); }
                     _converse.connection.reset();
                     this.render();
+                    if (_converse.registration_domain !== '') {
```

Comment:
I think this should rather check something like `!!_converse.registration_domain` so that other falsy values also work. For example, the above code won't work properly if the user passes in `null` or `undefined` or `false`.

The same applies for the other areas where you check specifically against an empty string.